### PR TITLE
Update to compile against latest Iron and Rust

### PR DESCRIPTION
**Pull request by AnthIste**
- Update `.call(req)` to `.handle(req)` to match the latest Iron
  - Add Rust features to suppress warnings: core, collections, path

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #![crate_name = "mount"]
 #![deny(missing_docs)]
 #![deny(warnings)]
-#![allow(unstable)]
+
+#![feature(core)]
```
Comment (reem) [inline src/lib.rs]: Can you collapse these to one line?

File: src/mount.rs [hunk 2]
```diff
@@ -89,7 +90,7 @@ impl Handler for Mount {
             // Search the Trie for the nearest most specific match.
             match self.inner.get_ancestor(key) {
                 Some(matched) => matched,
-                None => return Err(Box::new(NoMatch) as IronError)
+                None => return Err(IronError::new(NoMatch, status::Ok))
```
Comment (reem) [inline src/mount.rs]: This should be `status::NotFound`, since no handler was found.

**Comment by AnthIste** [comment]
I've made the changes you suggested. Do you want me to roll up the commits?

**Comment by ProtectedMode** [comment]
@reem, any progress on this and the Cargo package?

**Comment by AnthIste** [comment]
This was failing on Travis because of upstream dependencies but that all seems good now. @ProtectedMode you can always check out a working version locally and point to it with `.cargo/config`?

**Comment by reem** [comment]
@AnthIste @ProtectedMode I was blocked on merging this from travis, then I forgot about it and accidentally implemented the same fixes earlier today. Thanks for the PR, my bad managing this proactively.

**Comment by AnthIste** [comment]
@reem gotcha. I just tried rebasing onto master and got a bunch of merge conflicts, so I figured it was fixed :)

**Pull request closed without merging**